### DISCUSSION TITLE
Vercel deploy my source code of react instead the build

### DISCUSSION THREAD

**Discussion by mahmoudxyz** [vercel/community#3134] [Help] [answered]
### Summary

Vercel deployed my source code of react instead of the build hashed

The problem is that the source code of my react app can be seen from the browser instead of the hashed build of build after running "npm build"

I want the vercel to run the build folder instead of the source code

In "Build & Development Settings" I choose:

Framework and preset: Create React App

output directory: build

build command: npm run build

I also promote it to the production

### Steps to Reproduce

I want to make the production use only the build hashed of my react app

**Comment by swarnava** [comment]
Hi,

Do you have a repository which we can look at?

**Reply by mahmoudxyz** [reply]
It private project and not mine but I can still share some files. I tried it in my local machine and still the same. I think it's not vercel problem but still the problem there

I build the project and used this command

```
npm run build
serve -s build
```

After running the react app on the local server of the build, I still see the project source code instead of the hashed build

**Reply by swarnava** [reply]
Hi, you should run `npm run build` then `npm start` instead https://create-react-app.dev/docs/getting-started/#quick-start

**Reply by mahmoudxyz** [reply]
thanks but, running `npm run build` then `npm start` instead didn't solve the problem, I still see the source code from the browser

**Reply by swarnava** [reply]
Hi @mahmoudxyz , at this point we need see the source code to understand the issue.

**Reply by mahmoudxyz** [reply]
@swarnava  ok, can we arrange a meeting as soon as possible, and I can share the screen and try some solutions and then return back with a summary here?

**Reply by mahmoudxyz** [reply, marked as answer]
problem solved using this [article](https://dev.to/myogeshchavan97/how-to-hide-your-react-source-code-from-getting-displayed-in-chrome-dev-tools-when-deployed-to-production-41j7)  thanks man
